src/collection-view.js

```diff
@@ -151,6 +151,43 @@ Marionette.CollectionView = Marionette.AbstractView.extend({
     return this;
   },
 
+  // An efficient rendering used for filtering. Instead of modifying
+  // the whole DOM for the collection view, we are only adding or
+  // removing the related childrenViews.
+  setFilter: function(filter) {
+    if (!filter || this.filter === filter) {
+      return;
+    }
+    this.triggerMethod('before:filter', this);
```

Comment:
Yes, these events are new. I thought this can be useful if we want to listen the changes of filtering instead of 'classic' rendering. I will replace the current event names by yours.